community/kernel/src/main/java/org/neo4j/kernel/configuration/Settings.java

```diff
@@ -940,6 +940,54 @@ public String toString()
         };
     }
 
+    public static BiFunction<List<String>,Function<String,String>,List<String>> nonEmptyList =
+            new BiFunction<List<String>,Function<String,String>,List<String>>()
+            {
+                @Override
+                public List<String> apply( List<String> values, Function<String,String> settings )
+                {
+                    if ( values.isEmpty() )
+                    {
+                        throw new IllegalArgumentException( "setting must not be empty" );
+                    }
+                    return values;
+                }
+
+                @Override
+                public String toString()
+                {
+                    return "Non empty list";
```

Comment:
Should be `non` without the capital letter for consistency. I also believe that `nonempty` or `non-empty` would be more "currect" language wise.
Those two methods look really handy for future validation. :+1: